scripts/habitat_dataset_processing/habitat_dataset_processing/configs.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+
+class GroupType(Enum):
+    """
+    Hints how the assets should be packaged.
+    The assets in the generated 'assets.json' will be grouped according to this enum.
+    This maybe used by the client to package some assets in remote bundles to reduce build size.
+    """
+
+    LOCAL = 0
+    """
+    The assets should be packaged along with the build. This increases the build size (WebGL app, Android APK, etc.).
+    Use for assets that are expected to be used every time.
+    """
+
+    GROUP_BY_DATASET = 1
+    """
+    Group all assets within this data source together.
+    Use when most assets within a dataset are intended to be used together (e.g. a robot).
+    """
+
+    GROUP_BY_CHUNK = 2
+    """
+    Assets will be grouped in arbitrary chunks.
+    Use for large datasets with many objects that are occasionally used (e.g. ycb, ovmm, etc.).
+    """
+
+    GROUP_BY_SCENE = 3
+    """
+    Assets will be grouped by scenes.
+    Use for scene datasets (e.g. hssd).
+    """
+
+
+class Operation(Enum):
```

Comment:
This is a pretty rigid list. No way for a user script to do a custom operation without branching/modifying this package? E.g. I have one problematic model so I want to decimate it more aggressively, or remove it, or replace it with an alternate model. Or I want to apply a rotation to all assets due to some coordinate frame issue with the external engine for which I'm converting.

Reply:
I fully agree with you.

See my comment here: https://github.com/facebookresearch/habitat-lab/pull/2170#discussion_r2047244696